www/base/src/views/WorkersView/WorkersView.tsx

```diff
@@ -75,28 +81,46 @@ const getBuildsForWorkerMap = (workersQuery: DataCollection<Worker>,
 export const WorkersView = observer(() => {
   const accessor = useDataAccessor([]);
 
-  const showOldWorkers = buildbotGetSettings().getBooleanSetting("Workers.show_old_workers");
+  const settings = buildbotGetSettings();
+  const showOldWorkers = settings.getBooleanSetting("Workers.show_old_workers");
+
+  const [workerForActions, setWorkerForActions] = useState<null|Worker>(null);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [workerNameFilter, setWorkerNameFilter] = useState("");
+  const [showWorkersActions, setShowWorkersActions] = useState<boolean>(false);
 
   const workersQuery = useDataApiQuery(() => Worker.getAll(accessor, {query: {order: 'name'}}));
   const buildersQuery = useDataApiQuery(() => Builder.getAll(accessor));
   const mastersQuery = useDataApiQuery(() => Master.getAll(accessor));
-  const buildsQuery = useDataApiQuery(() =>
-    Build.getAll(accessor, {query: {
-        property: ["owners", "workername", "branch", ...getBuildLinkDisplayProperties()],
-        limit: 200,
-        order: '-buildid'
-      }
-    }));
-
-  const [workerForActions, setWorkerForActions] = useState<null|Worker>(null);
 
   const filteredWorkers = workersQuery.array.filter(worker => {
-    return isWorkerFiltered(worker, showOldWorkers);
+    return isWorkerFiltered(worker, showOldWorkers, workerNameFilter);
   }).sort((a, b) => a.name.localeCompare(b.name))
     .sort((a, b) => b.connected_to.length - a.connected_to.length);
 
-  const [showWorkersActions, setShowWorkersActions] =
-    useState<boolean>(false);
+  const [paginatedWorkers, paginationElement] = makePagination(
+    currentPage, setCurrentPage,
+    settings.getIntegerSetting("Workers.page_size"),
+    filteredWorkers
+  );
+
+  const paginatedWorkerIds = paginatedWorkers.map(w => w.workerid).sort();
+  const buildsQuery = useDataApiDynamicQuery(paginatedWorkerIds, () => {
+    // wait for workersQuery to be resolved to avoid querying withou
+    // workerid filter
+    if (!workersQuery.isResolved()) {
+      const col = new DataCollection<Build>();
+      col.resolved = true;
```

Comment:
I think col.resolved is not needed here. Until workersQuery is not resolved, builds query is not resolved either.

Reply:
If the 'real' query condition is never matched, the DataCollection will always be unresolved without doing this.
Which mean, in cases where `isResolved` is used to display a `LoadingSpan`, it will always wrongly be marked as 'loading'.

It's probably irrelevant here (with the current code anyway), but it became something of an habit to avoid the issue mentioned above.

Reply:
Thanks for explaining.